File: clients/node/examples/basic_uds.ts
// Minimal Varta beat loop — connect once over UDS, emit `Status.Ok`
// every 500ms.
//
// Mirror of `clients/python/examples/basic_uds.py` and
// `crates/varta-client/examples/basic.rs`. UDS is the preferred
// same-host transport: the observer reads kernel-attested peer
// credentials, classifying the source as `BeatOrigin::KernelAttested`
// and unlocking recovery-command eligibility.
//
// Requires the optional `node-unix-socket` addon to be installed
// (`npm install node-unix-socket`). On Windows or any platform
// without a prebuild, fall back to `basic_udp.ts`.
//
// Usage:
//   node --import tsx examples/basic_uds.ts [/path/to/varta.sock]

import { Varta, Status } from "../src/index.js";

const path = process.argv[2] ?? "/var/run/varta.sock";

const agent = Varta.connectUds(path);
console.log(`[varta] beating Ok every 500ms to uds://${path}`);

const timer = setInterval(() => {
  const outcome = agent.beat(Status.Ok);
  if (outcome.kind === "dropped") {
    console.warn(`[varta] dropped: ${outcome.reason}`);
  } else if (outcome.kind === "failed") {
    console.error(`[varta] failed: ${outcome.error.kind}`);
  }
}, 500);

const shutdown = (): void => {
  clearInterval(timer);
  agent.close();
  process.exit(0);
};
process.on("SIGINT", shutdown);
process.on("SIGTERM", shutdown);

File: clients/python/examples/basic_uds.py
"""Minimal Varta beat loop — connect once, emit Status.OK every 500 ms.

Run alongside varta-watch::

    varta-watch --socket /tmp/varta.sock --threshold-ms 2000 &
    python basic_uds.py /tmp/varta.sock
"""

from __future__ import annotations

import sys
import time

from varta import Status, Varta


def main(argv: list[str]) -> int:
    path = argv[1] if len(argv) > 1 else "/tmp/varta.sock"
    with Varta.connect(path) as agent:
        while True:
            outcome = agent.beat(Status.OK)
            if outcome.is_dropped:
                print(f"varta: beat dropped ({outcome.reason.value})", file=sys.stderr)
            elif outcome.is_failed:
                print(f"varta: beat failed: {outcome}", file=sys.stderr)
            time.sleep(0.5)


if __name__ == "__main__":
    sys.exit(main(sys.argv))
